### PR TITLE
[py-awkward1] add dlpack dependency

### PR DESCRIPTION
**Pull request by vvolkl**
Overlooked this when adding the new version

### CODE REVIEW

**Comment by adamjstewart** [comment]
It looks like the package name on PyPI is now `awkward`. The `awkward0` and `awkward1` names are still available for download, but are no longer getting new releases. Should we rename the Spack package too?

**Comment by vvolkl** [comment]
@adamjstewart agreed, now that would make sense. I'll do that.

By the way, I also had a github discussion with the package author about the patch for pybind11, and he didn't really intend for an external pybind11 to be used here. If we would use the source from pypi, the exact version he is using would be pulled in via a git submodule. I still prefer to have an external package, as long as it works with the patch.

**Comment by adamjstewart** [comment]
@vvolkl do you have time to do the package rename now or do you want to save that for another PR?

**Comment by adamjstewart** [comment]
The proper way to do the package rename would be to keep the old package but deprecate all of the versions.

**Comment by vvolkl** [comment]
@adamjstewart Yep, gave it a try -- let me know if that looks ok.

File: var/spack/repos/builtin/packages/py-awkward/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyAwkward(PythonPackage):
+    """ROOT I/O in pure Python and NumPy."""
+
+    git = "https://github.com/scikit-hep/awkward-1.0.git"
+    url = "https://github.com/scikit-hep/awkward-1.0/archive/0.3.1.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-awkward/package.py]: We should download from PyPI

File: var/spack/repos/builtin/packages/py-awkward/package.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyAwkward(PythonPackage):
+    """ROOT I/O in pure Python and NumPy."""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-awkward/package.py]: It looks like this description is from the uproot package, not awkward.

File: var/spack/repos/builtin/packages/py-awkward/package.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+    version('1.1.2', sha256='626e3a6a2a92dd67abc8692b1ebfa1b447b9594352d6ce8c86c37d7299dc4602')
+    version('0.3.1', sha256='7126d9feab8828b0b4f4c6dbc9e28c269a91e28eef4a6033d7ebb5db21f1dab3')
+
+    patch('pybind11.patch')
+
+    depends_on('py-setuptools', type='build')
+
+    depends_on('python@3.6:', type=('build', 'run'))
+    depends_on('py-numpy@1.13.1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-awkward/package.py]:
```suggestion
    depends_on('py-numpy@1.13.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-awkward/package.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    depends_on('py-pybind11')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-awkward/package.py]: What build type is this? Is it needed at build-time? Link-time? Run-time?

**Review by adamjstewart — COMMENTED** [review]
Might as well add the dlpack dependency to the deprecated package in case anyone is still using it.

File: var/spack/repos/builtin/packages/py-awkward/package.py (hunk 4 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-awkward/package.py]: I think `build`, `link` - I haven't actually tested it but scipy uses pybind11 that way too.

**Comment by vvolkl** [comment]
> Might as well add the dlpack dependency to the deprecated package in case anyone is still using it.

Done! Thanks for the comments.

File: var/spack/repos/builtin/packages/py-awkward/package.py (hunk 2 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-awkward/package.py]: Fixed

File: var/spack/repos/builtin/packages/py-awkward/package.py (hunk 1 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-awkward/package.py]: Done

File: var/spack/repos/builtin/packages/py-awkward/package.py (hunk 3 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-awkward/package.py]: Done

File: var/spack/repos/builtin/packages/py-awkward/package.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    git = "https://github.com/scikit-hep/awkward-1.0.git"
+    pypi = "awkward/awkward-1.1.2.tar.gz"
+    homepage = "https://awkward-array.org"
+
+    maintainers = ['vvolkl']
+
+    version('1.1.2', sha256='626e3a6a2a92dd67abc8692b1ebfa1b447b9594352d6ce8c86c37d7299dc4602')
+    version('0.3.1', sha256='7126d9feab8828b0b4f4c6dbc9e28c269a91e28eef4a6033d7ebb5db21f1dab3')
+
+    variant('builtin_pybind11', description="Build with the builtin version of pybind11", default=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-awkward/package.py]: I personally wouldn't use a variant for this, but it's your package so I'll let you decide

File: var/spack/repos/builtin/packages/py-awkward/package.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    """Manipulate JSON-like data with NumPy-like idioms."""
+
+    git = "https://github.com/scikit-hep/awkward-1.0.git"
+    pypi = "awkward/awkward-1.1.2.tar.gz"
+    homepage = "https://awkward-array.org"
+
+    maintainers = ['vvolkl']
+
+    version('1.1.2', sha256='626e3a6a2a92dd67abc8692b1ebfa1b447b9594352d6ce8c86c37d7299dc4602')
+    version('0.3.1', sha256='7126d9feab8828b0b4f4c6dbc9e28c269a91e28eef4a6033d7ebb5db21f1dab3')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-awkward/package.py]: These checksums need to be updated now, PyPI tarballs have different checksums than GitHub tarballs.

File: var/spack/repos/builtin/packages/py-awkward1/package.py [hunk 7]
```diff
@@ -25,5 +25,6 @@ class PyAwkward1(PythonPackage):
     depends_on('python@3.6:', type=('build', 'run'))
     depends_on('py-numpy@1.13.1:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-awkward1/package.py]: You can update these deptypes too

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-awkward/package.py (hunk 6 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-awkward/package.py]: Completely forgot about that, good catch!

File: var/spack/repos/builtin/packages/py-awkward1/package.py (hunk 7 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-awkward1/package.py]: Done

File: var/spack/repos/builtin/packages/py-awkward/package.py (hunk 5 above)
Comment (vvolkl) [reply var/spack/repos/builtin/packages/py-awkward/package.py]: Yes, I removed it again as I'm not sure about it as well. For now I'll try to make the pybind11 version more specific if there should be any version issues.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**